server/apps/microservices/src/processors/thumbnail.processor.ts

```diff
@@ -0,0 +1,37 @@
+import { Process, Processor } from '@nestjs/bull';
+import { Job } from 'bull';
+import { AssetEntity } from '@app/database/entities/asset.entity';
+import { Repository } from 'typeorm/repository/Repository';
+import { InjectRepository } from '@nestjs/typeorm';
+import sharp from 'sharp';
+
+@Processor('thumbnail-generator-queue')
+export class ThumbnailGeneratorProcessor {
+  constructor(
+    @InjectRepository(AssetEntity)
+    private assetRepository: Repository<AssetEntity>,
+  ) {}
+
+  @Process('generate-jpeg-thumbnail')
+  async generateJPEGThumbnail(job: Job) {
+    const { asset }: { asset: AssetEntity } = job.data;
+
+    console.log(asset);
+  }
+
+  @Process('generate-webp-thumbnail')
```

Comment:
What is the default concurrency for all these jobs? Generating the WebP thumbnails could still be quite intensive so probably don't want the background tasks going crazy and taking CPU, at least before it can be configurable for the user

Reply:
The concurrency for these jobs, if not specified, is set to the maximum allowable workers, which is the available Redis Client, I think it would make sense to set the level as 2 or so in this case

https://github.com/OptimalBits/bull/blob/develop/REFERENCE.md#queuegetworkers

Reply:
Yea I think setting some sensible limit would be good just so it doesn't eat CPU. Longer term I think these options arre great candidates for the admin settings page as some kind of advanced options.